src/item.cpp

```diff
@@ -218,6 +219,22 @@ item& item::deactivate( const Character *ch, bool alert )
     return *this;
 }
 
+item& item::activate()
+{
+    if( active ) {
+        return *this; // no-op
```

Comment:
What's the point in returning a reference to itself? This is not an operator and the value is never used. Even if it would be, it's generally better to separate getters (no side effects) from modifiers (no return values).